src/Symfony/Component/Security/Core/Authentication/Token/Storage/UsageTrackingTokenStorage.php

```diff
@@ -88,4 +88,19 @@ private function getSession(): SessionInterface
 
         return $this->container->get('request_stack')->getSession();
     }
+
+    private function shouldTrackUsage(): bool
+    {
+        if (!$this->enableUsageTracking) {
+            return false;
+        }
+
+        if (!$this->container->get('request_stack')->getMainRequest()) {
```

Comment:
We should also check if people inject the deprecated `session` service.
This check will also asserts the `request_stack` service exists.
```php
if ($this->container->has('session')) {
    return true;
}
```

Reply:
Thanks, added this snippet :)